tests/test_pages.py

```diff
@@ -0,0 +1,98 @@
+# pylint: disable=redefined-outer-name
+from __future__ import print_function
+
+from collections import namedtuple
+
+import pytest
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
+User = namedtuple('User', ['username', 'password', 'name'])
+
+PAGES = [
+    '/',
+    '/architectures',
+    '/hosts',
+    '/models',
+    '/media',
+    '/operatingsystems',
+    '/templates/ptables',
+    '/templates/provisioning_templates',
+    '/hostgroups',
+    '/common_parameters',
+    '/environments',
+    '/puppetclasses',
+    '/config_groups',
+    '/variable_lookup_keys',
+    '/puppetclass_lookup_keys',
+    '/smart_proxies',
+    '/compute_resources',
+    '/compute_profiles',
+    '/subnets',
+    '/domains',
+    '/http_proxies',
+    '/realms',
+    '/locations',
+    '/organizations',
+    '/auth_source_ldaps',
+    '/users',
+    '/usergroups',
+    '/roles',
+    '/bookmarks',
+    '/settings',
+    '/about',
+]
+
+
+@pytest.fixture
+def firefox_options(firefox_options):
+    firefox_options.headless = True
+    return firefox_options
+
+
+@pytest.fixture
+def chrome_options(chrome_options):
+    chrome_options.headless = True
+    # The error log is a non-w3c option
+    chrome_options.add_experimental_option('w3c', False)
+    return chrome_options
+
+
+@pytest.fixture
+def user(variables):
+    return User(variables.get('username', 'admin'), variables.get('password', 'changeme'),
+                variables.get('name', 'Admin User'))
+
+
+@pytest.fixture(params=PAGES)
+def url(request, base_url):
+    if not base_url:
+        raise ValueError('Base URL not specified')
+    return base_url + request.param
+
+
+@pytest.mark.nondestructive
+@pytest.mark.selenium
+def test_page(selenium, user, url):
```

Comment:
just that I understand it correctly. this will be executed for every entry in `PAGES` and expects the following flow:
1. the user is redirected to the login page
1. the user logs in and is redirected to the real page
1. the user is properly logged in (the full name is checked) and the real page renders correctly?

Reply:
Correct. The selenium fixture spawns a new browser for every test (which also means that it can run concurrently). That's why the login is needed: there is no state prior to the test.

Reply:
How about turning the login into a fixture itself?
- also, i see the `user` fixture uses the same user (admin). Just a word of warning for the future - if your tests will involve work with contexts (orgs/logs) in the future, you might need to have separate users for each session, as the context seems to be sticky for the user (if you change the context for admin in session#1, it will be also changed for admin user in session#2).

Reply:
I thought about that, but thought it was a case of premature optimization. Currently it's a single function so very easy to understand. I don't know how many more UI tests will show up. I'd be more interested in using the API.